File: java/src/test/java/org/codemash/RomanNumeralTest.java
package org.codemash;

import static org.junit.Assert.*;
import org.junit.*;

public class RomanNumeralTest {
  private RomanNumeral numeral;

  @Before
  public void setup() {
    numeral = new RomanNumeral();
  }

  @Test
  public void converts1ToI() {
    assertEquals("I", numeral.convert(1));
  }    

  @Test
  public void converts2ToII() {
    assertEquals("II", numeral.convert(2));
  }    

  @Test
  public void converts3ToIII() {
    assertEquals("III", numeral.convert(3));
  }    

  @Test
  public void converts4ToIV() {
    assertEquals("IV", numeral.convert(4));
  }    

  @Test
  public void converts5ToV() {
    assertEquals("V", numeral.convert(5));
  }    

  @Test
  public void converts6ToVI() {
    assertEquals("VI", numeral.convert(6));
  }    

  @Test
  public void converts9ToIX() {
    assertEquals("IX", numeral.convert(9));
  }    

  @Test
  public void converts10ToX() {
    assertEquals("X", numeral.convert(10));
  }    
}

File: js/test/roman_numeral_test.js
require('should');
var RomanNumeral = require(__dirname + '/../lib/roman_numeral');

describe('RomanNumeral', function() {
  var numeral = new RomanNumeral();
  it('1 to I', function() {
    numeral.convert(1).should.equal('I');
  });
  it('2 to II', function() {
    numeral.convert(2).should.equal('II');
  });
  it('3 to III', function() {
    numeral.convert(3).should.equal('III');
  });
  it('4 to IV', function() {
    numeral.convert(4).should.equal('IV');
  });
  it('5 to V', function() {
    numeral.convert(5).should.equal('V');
  });
  it('6 to VI', function() {
    numeral.convert(6).should.equal('VI');
  });
  it('9 to IX', function() {
    numeral.convert(9).should.equal('IX');
  });
  it('10 to X', function() {
    numeral.convert(10).should.equal('X');
  });
});

